### PR TITLE
Defect/dnn 21619

### PR DESCRIPTION
**Pull request by mean2me**
## Summary
Allow possibility to create a ContentItem with a specified "CreatedByUserID" value.
Feature allowed to super users only, TBD if we can remove this restriction.

### CODE REVIEW

**Comment by sleupold** [comment]
is there a concrete use case for this change?

**Comment by tpluscode** [comment]
Please see #2223 @sleupold

**Comment by sleupold** [comment]
ok, thanks for the link. I agree, this is a bug that needs to be fixed.

File: DNN Platform/Library/Entities/Content/ContentController.cs [hunk 1]
```diff
@@ -60,11 +60,16 @@ public int AddContentItem(ContentItem contentItem)
...
-	        var userId = UserController.Instance.GetCurrentUserInfo().UserID;
-            contentItem.ContentItemId = _dataService.AddContentItem(contentItem, userId);
-            contentItem.CreatedByUserID = userId;
-            contentItem.LastModifiedByUserID = userId;
+            var currentUser = UserController.Instance.GetCurrentUserInfo();
+            var createdByUserId = currentUser.UserID;
+            if(contentItem.CreatedByUserID != currentUser.UserID && currentUser.IsSuperUser)
+            {
+                createdByUserId = contentItem.CreatedByUserID;
+            }
```
[the remark this answers is not in the document]
Comment (sleupold) [reply DNN Platform/Library/Entities/Content/ContentController.cs]: Who is allowed to update a contentitem of another user?
at least an admin, but in many use cases a supervisor as well - but this should be restricted by the module, not the core API, IMO

[the remark this answers is not in the document]
Comment (bdukes) [reply DNN Platform/Library/Entities/Content/ContentController.cs]: I agree, this doesn't need to be restricted by the API, but by the extension using the API

[the remark this answers is not in the document]
Comment (mean2me) [reply DNN Platform/Library/Entities/Content/ContentController.cs]: I agree it's not the right place but good it gave us opportunity to talk about it. So, I'm gonna remove this check but we should consider some refactoring of this part, IMO.

**Pull request merged**